src/actions/UploadActions.tsx

```diff
@@ -1,47 +1,63 @@
 import { uploadFile } from '../api/upload';
 import { GenericAction } from '../models/action';
+import { Upload } from '../models';
 
 export const ActionTypes = {
     UPLOAD_REQUEST: 'UPLOAD_REQUEST',
     UPLOAD_PROGRESS: 'UPLOAD_PROGRESS',
-    UPLOAD_CLEAR: 'UPLOAD_CLEAR'
+    SELECT_UPLOAD_FILE: 'SELECT_UPLOAD_FILE'
 };
 
-export const uploadRequest = (
-    customerId: string,
-    file: File,
-    config = {}
-): GenericAction => {
+export const uploadRequest = (upload: Upload, config = {}): GenericAction => {
     const formData = new FormData();
-    formData.append('file', file, file.name);
+    formData.append('file', upload.file, upload.file.name);
+    formData.append('reportName', upload.reportName);
+
+    if (upload.reportDescription) {
+        formData.append('reportDescription', upload.reportDescription);
```

Comment:
It's interesting that we have to check each value in `upload` and call `formData.append` for each one... i bet there is a way we can simplify this to a single call that converts `upload` into a `FormData` object.

Maybe something like:
```
const objectToFormData = obj => {
  const formData = new FormData();
  Object.keys(obj).forEach(key => formData.append(key, obj[key].toString()));
  return formData;
};
```

And then you can just `const formData = objectToFormData(upload);`. I wonder if there is something similar built-in to the FormData prototype that we could use instead, though.

Reply:
Thanks for the suggestion. This is a very good one which will be included in this PR